### PR TITLE
feat: Support for @testing-library/jest-native

### LINKED ISSUE #298: Make sure we support @testing-library/jest-native
## Describe the Feature
Current set of jest matchers is limited only to the base ones provided by jest. This makes simple queries like checking the text content of found element rather non-intuitive be directly working with its `children` prop.

The @testing-library/jest-native adds a number of useful React Native specific queries like: toHaveTextContent, toHaveStyle, toBeEnabled/Disabled, etc. 

Such addition is motivated by providing the best potential developer experience for RNTL users. 

As a reference point, the original React Testing Library includes @testing-library/jest-dom dev dependency.

## Possible Implementations
Add @testing-library/jest-native to dev dependencies. 

Alternatively, we can document existence of this related package and provide a suggestion to use it with RNTL as a suggested practice.

## Related Issues
None.

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Added support for @testing-library/jest-native matchers by providing unit tests in our project, as well as documentation for setting up these matchers.

Fixes #298 

### Test plan
Added unit tests for @testing-library/jest-native matchers. Run `yarn test`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,12 +64,42 @@ You can find the source of `QuestionsBoard` component and this example [here](ht
...
+```sh
+yarn add --dev @testing-library/jest-native
+```
+
+#### Using `npm`
+```sh
+npm install --save-dev @testing-library/jest-native
+```
+
+Then add following line to your jest.config.js:
```
Comment (thymikee) [inline README.md:93]:
```suggestion
Then add the following line to your jest.config.js:
```

File: src/__tests__/jest-native.test.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+// @flow
+/* eslint-disable react-native/no-color-literals */
```
Comment (thymikee) [inline src/__tests__/jest-native.test.js:2]: we can disable it globally if you like

File: src/__tests__/jest-native.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    backgroundColor: 'green',
+  });
+  expect(getByA11yHint('Empty Text')).not.toHaveStyle({ color: 'green' });
+  expect(getByA11yHint('Empty Text')).not.toHaveStyle({
+    color: 'green',
+    backgroundColor: 'green',
+  });
+});
+
+test.skip('currently failing jest-native matchers tests', () => {
```
Comment (thymikee) [inline src/__tests__/jest-native.test.js:71]: would be nice to annotate what exactly fails and why

Comment (mdjastrzebski) [reply src/__tests__/jest-native.test.js]: Done

File: README.md [hunk 4]
```diff
@@ -64,12 +64,46 @@ You can find the source of `QuestionsBoard` component and this example [here](ht
...
 
+#### Using `npm`
+
+```sh
+npm install --save-dev react-native-testing-library
+```
+
 This library has a peerDependencies listing for `react-test-renderer` and, of course, `react`. Make sure to install them too!
 
+### Additional jest matchers
```
Comment (thymikee) [inline README.md:81]:
```suggestion
### Additional Jest matchers
```

File: website/docs/GettingStarted.md [hunk 5]
```diff
@@ -50,10 +50,42 @@ You can find the source of `QuestionsBoard` component and this example [here](ht
...
+
+```sh
+npm install --save-dev react-native-testing-library
+```
+
 This library has a peerDependencies listing for `react-test-renderer` and, of course, `react`. Make sure to install them too!
 
 As you may have noticed, it's not tied to React Native at all – you can safely use it in your React components if you feel like not interacting directly with DOM.
+
+### Additional jest matchers
```
Comment (thymikee) [inline website/docs/GettingStarted.md:69]:
```suggestion
### Additional Jest matchers
```

**Comment by mdjastrzebski** [comment]
@cross19xx This seems to be actually @testing-library/jest-native issue with incomplete flow typings, as they are inferred by flow, and not exported by library. So the code works correctly in runtime, but `yarn flow-check` complains.

**Pull request merged**